autoscale-core/src/main/java/com/hpe/caf/autoscale/core/GovernorImpl.java

```diff
@@ -43,16 +50,51 @@ public GovernorImpl(final int stageOneLimit, final int stageTwoLimit, final int
         this.stageOneShutdownPriorityLimit = stageOneLimit;
         this.stageTwoShutdownPriorityLimit = stageTwoLimit;
         this.stageThreeShutdownPriorityLimit = stageThreeLimit;
+        this.scalerThreads = new HashMap<>();
+    }
+
+    @Override
+    public boolean freeUpResourcesForService(final String serviceRef)
+    {
+        final AdvancedInstanceInfo lastInstanceInfo = instanceInfoMap.getOrDefault(serviceRef, null);
+        if (lastInstanceInfo == null) {
```

Comment:
What is the purpose of this check?  Why does `lastInstanceInfo` come into this?  Probably best to talk me through this function.

Reply:
The default if the item is not in the map of instanceInfos is to return null. This same check is also done at the top of govern().